src/metabase/util/malli/fn.clj

```diff
@@ -189,41 +213,43 @@
   (validate error-context schema value ::invalid-output)
   value)
 
-(defn- varargs-schema? [[_cat & args :as _input-schema]]
-  (letfn [(star-schema? [schema]
-            (and (sequential? schema)
-                 (= (first schema) :*)))]
-    (star-schema? (last args))))
+(defn- varargs-type [input-schema]
+  (-> input-schema meta :varargs/type))
 
 (defn- input-schema-arg-names [[_cat & args :as input-schema]]
-  (let [varargs?    (varargs-schema? input-schema)
-        normal-args (if varargs?
-                      (butlast args)
-                      args)]
+  (let [varargs-type (varargs-type input-schema)
+        normal-args  (if varargs-type
+                       (butlast args)
+                       args)]
     (concat
      (for [n (range (count normal-args))]
        (symbol (str (char (+ (int \a) n)))))
-     (when varargs?
-       ['more]))))
+     (case varargs-type
+       :varargs/sequential ['more]
+       :varargs/map        ['kvs]
+       nil))))
```

Comment:
TIL: clojure's `case` does not return `nil` by default

Reply:
`No matching clause` is a famous exception.